coffea/nanoevents/methods/vector.py

```diff
@@ -136,13 +152,30 @@ def multiply(self, other):
             with_name="TwoVector",
         )
 
+    @awkward.mixin_class_method(numpy.divide, {numbers.Number})
+    def divide(self, other):
```

Comment:
Could this be implemented by calling multiply on the reciprocal?

Reply:
Would this not be a performance loss? `other` can be an array of numbers, thus it would first allocate an array for the reciprocal and then another one for the multiplication. And I could imagine reciprocal & multiplication also takes more cycles than just a division.

Reply:
In general multiplication is much cheaper than division. So by computing the reciprocal and re-using it for each cartesian component multiplication should actually increase performance. The extra memory cost can be avoided by overwriting, e.g. `other = 1 / other`

Reply:
I'm not sure how reference lifecycle behaves in a call stack, but python may do "the right thing" if you just implement this as `self.multiply(1 / other)`

Reply:
The only difference between any of the `divide` methods would then by their documentation string. As everything inherits from `TwoVector` it would be enough to implement it only there. Do you want to keep the individual implementations?

Reply:
Sure, with a generic enough docstring just the TwoVector impl. is fine